emData/generate_TB.py

```diff
@@ -122,9 +122,9 @@ class ITC(Enum):
         "template<TF::seed Seed> constexpr int getNumDiskFMMemPerStub();\n"
         "template<TF::seed Seed> constexpr int getNumBarrelStub();\n"
         "template<TF::seed Seed> constexpr int getNumDiskStub();\n"
-        "template<unsigned Seed> int getMPARNPages(const ITCType &);\n"
-        "template<unsigned Seed> int getMPARMem(const ITCType &);\n"
-        "template<unsigned Seed> int getMPARPage(const ITCType &);\n"
+        "template<TF::seed Seed> inline    int getMPARNPages(const ITCType &);\n"
+        "template<TF::seed Seed> inline    int getMPARMem(const ITCType &);\n"
+        "template<TF::seed Seed> inline    int getMPARPage(const ITCType &);\n"
```

Comment:
@tomalin Probably it's better to just make these `constexpr`, as that should replace the function calls with their return values at compile time. IIRC, this would require changing the input argument to something simple like `unsigned` or `unsigned short`, but that shouldn't cause any problems on the FW side.

```suggestion
        "template<TF::seed Seed> constexpr int getMPARNPages(const unsigned short);\n"
        "template<TF::seed Seed> constexpr int getMPARMem(const unsigned short);\n"
        "template<TF::seed Seed> constexpr int getMPARPage(const unsigned short);\n"
```

Reply:
They can't be constexpr. For example:

template<> inline int
getMPARNPages<TF::L1L2>(const ITCType &iTC) {
  return (iTC == 5 || iTC == 6) ? 1 : (
         (iTC == 3 || iTC == 4 || iTC == 7 || iTC == 8) ? 2 : 3);
}

depends on const ITC, whose value is not known at compile time.

Reply:
The arguments to a `constexpr` function don't need to be known at compile time, the function just needs to return a constant value after argument substitution, whether that happens at compile time or run time.

But your point is valid, since the function cannot be evaluated at compile time if the arguments aren't known. So I think making the functions `inline` instead makes sense.